serde/src/tagger.rs

```diff
@@ -0,0 +1,23 @@
+/// Provides the correct tag for a given format.
+pub trait Tagger {
```

Comment:
One needs to keep in mind that many applications support multiple serialization formats. They may have colliding tags. For this reason the serialization format needs to be checked. One example is a deserializer that encounters an integer tag 0. In BSON it describes a generic binary value while in CBOR it is used for a datetime  value. The function you propose does not handle this ambiguity.

In serialization this problem is even worse: you have a tag 42 for RGBA colors used in CBOR but it must never be used with BSON.